### DISCUSSION TITLE
How to pass Lazy input to ShellCommandTask?

### DISCUSSION THREAD

**Discussion by adi611** [nipype/pydra#663] [Q&A]
@djarecka - I am attempting to implement the worker from the PSI/J site, which consists of two parts: 1. Extracting text from a PDF (`pdftotext INPUT.pdf OUTPUT.txt`) and 2. Creating a word cloud image from the extracted text (`wordcloud_cli --text OUTPUT.txt --imagefile IMAGE.png`).

My approach involves creating `ShellCommandTasks` for the above commands and adding them to the workflow to execute them. Here is my working code:
```
cmd1 = 'pdftotext'
args = ['/content/Lorem_ipsum.pdf', '/content/Output.txt']

shelly1 = pydra.ShellCommandTask(name='shelly1', executable=cmd1, args=args)
print('cmndline = ', shelly1.cmdline)

cmd2 = 'wordcloud_cli'
args2 = ['--text', '/content/Output.txt', '--imagefile', '/content/Image.png']

shelly2 = pydra.ShellCommandTask(name='shelly2', executable=cmd2, args=args2)
print('cmndline = ', shelly2.cmdline)

wf1 = pydra.Workflow(name = 'wf1', input_spec=['x'], x=3)
wf1.add(shelly1)
wf1.add(shelly2)

wf1.set_output([('out_s1', wf1.shelly1.lzout.stdout), ('out_s2', wf1.shelly2.lzout.stdout)])

with pydra.Submitter(plugin='cf') as sub:
    sub(wf1)

wf1.result()
```

One observation I made is that it is necessary to provide the `input_spec` and `set_output` to the workflow. In the given code, I provided a dummy `input_spec` because I only wanted to execute static shell commands where the filenames were already defined in the `args` parameter. However, let's say I do want to provide my PDF filename as input to the workflow. In that case, how can I use it as a 'Lazy Input' for my `shelly1` task?

**Comment by djarecka** [comment]
you can point argument of your task to workflow input like this `wf.lzin.x`, see example [here](https://nipype.github.io/pydra-tutorial/notebooks/4_intro_workflow.html)

An example that is closely matching  your case you can find in the tests, e.g. [here](https://github.com/nipype/pydra/blob/426564eca3efa7d840fdca33bdbef6a1c88e2ffc/pydra/engine/tests/test_shelltask.py#L1556)
